### DISCUSSION TITLE
memory utilisation issue comes after implementation fflate installation in our different portal

### DISCUSSION THREAD

**Discussion by sannysoni123** [101arrowz/fflate#88] [Q&A]
Different browser, we have checked download zip in our portal like chrome, Firefox & Edge but memory crashed sometime on edge and all browser hang sometime. should we consider large number of MB files make this happened?

code is simple that we have using.. please guide me on this..
```
downloadMedia(mediaFiles, zipName) {
        // remove duplicate file
        if (mediaFiles && mediaFiles.length > 0) {
            mediaFiles = mediaFiles.filter((value, index) =>
                mediaFiles.findIndex(x => x.name === value.name) === index
            );
            Promise.all(mediaFiles.map(file => new Promise(resolve => { // promise all for media files
                this.getFileFromUrl(file.url).subscribe(async (data: any) => { // async data from url
                    const filename = file.name;
                    resolve([filename, new Uint8Array(await data.arrayBuffer())]);
                });
            }))).then(files => {
                // Higher level means lower performance but better compression
                // The level ranges from 0 (no compression) to 9 (max compression)
                // The default level is 6
                const zip = fflate.zipSync((<any>Object).fromEntries(files), { level: 0 }); // level zero
                const url = URL.createObjectURL(new Blob([zip]));
                const downloadLink = document.createElement('a');
                downloadLink.href = url;
                downloadLink.download = `${zipName}.zip`;
                downloadLink.click();
                URL.revokeObjectURL(url); // revoke object url
            });
        }
    }
```

**Comment by 101arrowz** [comment]
`fflate` has no memory overhead for ZIP objects besides the input and output buffers. How large are the ZIP files you are generating? If they are hundreds of megabytes large then it's something you can only avoid with data streaming, I can give you some help on that regard but if you send me an example of the ZIP files you are generating it would be helpful.

**Reply by sannysoni123** [reply]
it would be grateful if you help me out on this
- not using data streaming
- trying to generate upto 100MB
- is there any browser specific limitation?

**Reply by 101arrowz** [reply]
100MB will be fine if your browser is new and can use 200MB of memory. You basically need to double the memory of the ZIP file if you don't stream the data. Whereas with data streaming it could be only 100kB to 1MB of memory used.
